File: Java/2 - WeatherStation (Observer)/src/main/java/WeatherData.java

import java.util.ArrayList;
import java.util.List;

/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

/**
 *
 * @author ismas
 */
public class WeatherData implements Subject {
    private List<Observer> observers;
    private float temperature;
    private float humidity;
    private float pressure;
    
    public WeatherData(){
        observers = new ArrayList<Observer>();
    }
    
    public void registerObserver( Observer observer ){
        observers.add( observer );
    }
    
    public void removeObserver( Observer observer ){
        observers.remove( observer );
    }
    
    public void notifyObservers(){
        for( Observer observer : observers ){
            observer.update(temperature, humidity, pressure);
        }
    }
    
    public float getTemperature(){
        return temperature;
    }
    
    public float getHumidity(){
        return humidity;
    }
    
    public float getPressure(){
        return pressure;
    }
    
    public void measurementsChanged(){
        notifyObservers();
    }
    
    public void setMeasurements( float temperature, float humidity, float pressure ){
        this.temperature = temperature;
        this.humidity = humidity;
        this.pressure = pressure;
        measurementsChanged();
    }
}

File: C#/2. WeatherStation (Observer)/Classes/WeatherData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using WeatherStation.Interfaces;

namespace WeatherStation.Classes
{
    public class WeatherData
    {
        private List<IObserver> _Observers;
        private float _Temperature;
        private float _Humidity;
        private float _Pressure;

        public WeatherData()
        {
            _Observers = new List<IObserver>();
        }
        public void RegisterObserver(IObserver observer)
        {
            _Observers.Add(observer);
        }

        public void RemoveObserver(IObserver observer)
        {
            _Observers.Remove(observer);
        }

        public void NofifyObservers()
        {
            foreach (IObserver observer in _Observers)
            {
                observer.Update(_Temperature, _Humidity, _Pressure);
            }
        }

        public void MeasurementsChanged()
        {
            NofifyObservers();
        }

        public void SetMeasurements(float temperature, float humidity, float pressure)
        {
            _Temperature = temperature;
            _Humidity = humidity;
            _Pressure = pressure;
            MeasurementsChanged();
        }
    }
}
